src/main/java/org/numenta/nupic/network/Layer.java

```diff
@@ -1909,12 +1913,35 @@ private void clearSubscriberObserverLists() {
      * @return
      */
     NamedTuple makeClassifiers(MultiEncoder encoder) {
+        Map inferredFields = (Map<String, Class<? extends Classifier>>) params.get(KEY.INFERRED_FIELDS);
+        if(inferredFields == null || inferredFields.entrySet().size() == 0) {
+            throw new IllegalStateException(
+                    "KEY.AUTO_CLASSIFY has been set to \"true\", but KEY.INFERRED_FIELDS is null or\n\t" +
+                    "empty. Must specify desired Classifier for at least one input field in\n\t" +
+                    "KEY.INFERRED_FIELDS or set KEY.AUTO_CLASSIFY to \"false\"."
```

Comment:
`KEY.AUTO_CLASSIFY` is [set to `false` by default](https://github.com/numenta/htm.java/blob/master/src/main/java/org/numenta/nupic/Parameters.java#L139). So simply not modifying the default value will remedy this exception. The key should only ever be true or false, unless the user explicitly sets it to null.

So while I think this exception message is _accurate_, it may not necessarily be _helpful_. I will change this, please see what you think of it after I update it.